Core/MaterialParameters.h

```diff
@@ -20,40 +20,93 @@
 #include <G4SystemOfUnits.hh>
 #include <G4ThreeVector.hh>
 #include <globals.hh>
+#include <vector>
+
+struct MaterialConstants
+{
+  MaterialConstants(std::vector<G4double> oPsLF, std::vector<G4double> oPsProb, 
+                    G4double pPsLF, G4double pPsFrac, 
+                    std::vector<G4double> DirLF, std::vector<G4double> DirProb);
+  std::vector<G4double> oPsLifetimes;
+  std::vector<G4double> oPsProbabilities;
+  G4double pPsLifetime;
+  G4double pPsFraction;
+  std::vector<G4double> directLifetimes;  
+  std::vector<G4double> directProbabilities;
+};
 
 class MaterialParameters
 {
 public:
-  //! Mean lifetime denoted as Tau (greek letter)
+  static const MaterialConstants XAD4;
+  static const MaterialConstants Al;
+  static const MaterialConstants Kapton;
+  static const MaterialConstants Plexiglass; // PMMA
+  static const MaterialConstants Scin;
+  static const MaterialConstants PA6;
+  static const MaterialConstants Air;
+  static MaterialConstants Temp;
+  
+  MaterialParameters();
+  MaterialParameters(std::vector<G4double> oPsLF, std::vector<G4double> oPsProb, 
```

Comment:
If you pass the vectors that are read-only parameters, then it is better to do:
const std::vector<G4double>& oPsLF, ...